jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -569,6 +573,13 @@ public GPUAssist run() {
 }
 /*[ENDIF] JAVA_SPEC_VERSION >= 9 */
 
+/*[IF JAVA_SPEC_VERSION >= 24]*/
+static URL codeSource(Class<?> callerClass) {
+	CodeSource cs = callerClass.getProtectionDomainInternal().getCodeSource();
+	return (cs == null) ? null : cs.getLocation();
```

Comment:
Please don't abbreviate unnecessarily; suggest `cs` should be `codeSource`.

Reply:
Updated